extensions/amp-jwplayer/0.1/amp-jwplayer.js

```diff
@@ -291,38 +311,54 @@ class AmpJWPlayer extends AMP.BaseElement {
     this.contentSearch_ = element.getAttribute('data-content-search') || '';
     this.contentBackfill_ = element.getAttribute('data-content-backfill') || '';
     this.contentRecency_ = element.getAttribute('data-content-recency') || '';
+    this.queryString_ = element.getAttribute('data-player-querystring') || '';
 
     installVideoManagerForDoc(this.element);
     Services.videoManagerForDoc(this.element).register(this);
   }
 
   /** @override */
   layoutCallback() {
-    const queryParams = dict({
-      'search': this.getContextualVal_() || undefined,
-      'recency': this.contentRecency_ || undefined,
-      'backfill': this.contentBackfill_ || undefined,
-      'isAMP': true,
-    });
+    return this.getConsentData_().then(() => {
+      const queryParams = dict({
+        'search': this.getContextualVal_() || undefined,
+        'recency': this.contentRecency_ || undefined,
+        'backfill': this.contentBackfill_ || undefined,
+        'isAMP': true,
+        'consentState': this.consentState_ || undefined,
+        'consentValue': this.consentString_ || undefined,
+        'consentGdpr': this.consentMetadata_
+          ? this.consentMetadata_.gdprApplies
+          : undefined,
+      });
+
+      const url = this.getSingleLineEmbed_();
+      let src = addParamsToUrl(url, queryParams);
+      src = addParamsToUrl(
+        src,
+        getDataParamsFromAttributes(this.element, null, /^playerParam(.+)/)
+      );
+      if (this.queryString_) {
+        src += `&${this.queryString_}`;
```

Comment:
@alanorozco Do we have a pattern preference here? Concern is this arbitrarily adds an `&` separator between query elements.

Reply:
Hopefully this shouldn’t append an `&` arbitrarily since it only appends it if there is also a queryString value, but open if there is a preferable pattern.

Reply:
The concern would be if src didn't already have query parameters, today or in a future refactor.

I think this is fine for now, but might be worth a comment.